### ISSUE TITLE
IntelliJ code flow analysis confused by `io.quarkus.logging.Log` stubs

### ISSUE THREAD

**Issue by Felk** [quarkusio/quarkus#25419]
### Description

When using `io.quarkus.logging.Log` for logging, the code flow analysis of IntelliJ occasionally gets a bit confused, because the implementation stubs of `io.quarkus.logging.Log` throw exceptions:
```java
public static void tracev(String format, Object... params) {
    throw fail();
}
```

For example the following code:
```java
if (someBool)
    Log.tracev("the bool is true");
// some other code
if (someBool)
    System.out.println("doing the thing...");
```

shows the following warning "Condition 'someBool' is always 'false'":
![grafik](https://user-images.githubusercontent.com/1996138/167130635-512673e5-f5e8-49fe-aa6e-639731b5a137.png)

### Implementation ideas

It's possible to avoid this issue by adding a condition to the stubs that is de-facto always true, but not provably always true by static code analysis, e.g.:
```java
public static void tracev(String format, Object... params) {
    if (System.nanoTime() >= 0) throw fail(); // need runtime condition, otherwise static code flow analysis assumes this method always throws
}
```

**Comment by Ladicek** [comment]
That makes sense to me, but `System.nanoTime() >= 0` is _not_ "de-facto always" true. `System.currentTimeMillis() >= 0` should be, unless your computer has a very wrongly configured time (which may happen I guess?). I think I'd much rather come up with an expression that would be complex enough for the control flow analysis to give up, or perhaps add `@Contract`... Need to think about it.

**Comment by Ladicek** [comment]
It seems replacing `throw fail();` with `if (always()) throw fail(); else throw fail();`, where `always()` is another `static` method that always returns `true`, is enough to fool the control flow analysis algorithm in IntelliJ :-) Will submit a PR shortly.

**Comment by Ladicek** [comment]
@Felk fix is in #25448. If you have a chance to verify (by building Quarkus from my branch), I'd appreciate that. But it seems to help in my IntelliJ, which is 2022.1.
